### ISSUE TITLE
onMapReady not always called with google maps on latest commit in master

### ISSUE THREAD

**Issue by Shaninnik** [react-native-maps/react-native-maps#4066]
# Bug report

## Summary

I am trying to use the lib from commit 22a9c9b6e38c3845ed3ab61b4da32a24000efe6c to get it running on M1 mac simulator, and it does, but now `onMapReady` is not always called (it is extremely sporadic). I know that this is not yet released but probably worth looking at.

## Environment info

"react-native": "0.66.3"
"react": "17.0.2"
"react-native-maps": "https://github.com/react-native-maps/react-native-maps.git#22a9c9b6e38c3845ed3ab61b4da32a24000efe6c"

I am using GoogleMaps.

Target Platform: eg: iOS - 12.0 Simulator

## Steps to reproduce

1. Create new project and install react-native-maps "https://github.com/react-native-maps/react-native-maps.git#22a9c9b6e38c3845ed3ab61b4da32a24000efe6c"
2. Render map with google provider:
```
<MapView
      style={[styles.map]}
      provider={'google'}
      onMapReady={() => console.log('Map ready')}></MapView>
```
3. Launch app and observe 'Map ready' logged in the console (or not)
4. Open react dev menu, hit reload, and see that 'Map ready' is logged in the console (or not)
5. Repeat step 4 until you see 'Map ready' not logged

On bare minimum example I sometimes need to reload several times to see missing `onMapReady`. In complex real life app with conditional rendering I almost never see `onMapReady` called.

Describe what you expected to happen:

`Map ready` printed on each start and reload.

**Comment by christopherdro** [comment]
@bhishaksanyal

**Comment by deng-yc** [comment]
Same problem

**Comment by verochan** [comment]
I'm suffering the same problem on a Macbook Pro mid 2012 on XCode iOS simulator since v0.30.0 and using GoogleMaps. On the other hand, on Android emulator, iOS & Android devices, `onMapReady` callback always gets called.

This is really frustrating, as I was trying to solve this [problem](https://github.com/react-native-maps/react-native-maps/pull/4036) (which is not still merged into master branch), and when using this branch on my project, I discover that, besides, v0.30.0 breaks the `onMapReady` callback on iOS simulators.

**Comment by Shaninnik** [comment]
When I was testing it, for some reason, it was happening much more frequent on my real project and less frequent on clean project, so might be possible that you are just not seeing it. Also might be possible that it works fine now with new master and 0.67.2 react. I will try it again tomorrow to see if I can still replicate.

**Comment by Simon-TechForm** [comment]
@Shaninnik have you had the time to look into this again?

**Comment by Shaninnik** [comment]
@Simon-TechForm same issue, tried from latest commit https://github.com/react-native-maps/react-native-maps/commit/3338a960c62cc3c364286433ab98e5cf59435ebf and no luck, onMapReady not called, was testing on device

**Comment by Shaninnik** [comment]
I am trying to dig deeper, it is quite a lot of code, but I've added some NSLogs and stepped in with debugger through AIRGoogleMap.m and found that in `(void)didPrepareMap` `self.onMapReady` is always null and thus never called. Just to verify I've tried the same with `onMapLoaded` and in `(void)mapViewDidFinishTileRendering` both `self.onMapReady` and `self.onMapLoaded` are already not null and `onMapLoaded` called properly. So it seems like after the google maps update `didPrepareMap` happens before React Native bridges all props.

**Comment by Simon-TechForm** [comment]
@Shaninnik thank you! I will take a look again.

**Comment by Simon-TechForm** [comment]
@Simon-TechForm notify

**Comment by Simon-TechForm** [comment]
As you might have guessed from the comment above, I've just published a beta release that should fix the issue. I would love to get feedback from anyone able to try it out 🤞

**Comment by MadeinFrance** [comment]
It works in DEBUG mode 🥳
-> Thanks for looking into that @Simon-TechForm 
I would go for a minor release 🚀

**Comment by BrodaNoel** [comment]
We kind of have a problem here.
This bug is really important, and it's present in version `0.31.1`, which is the compatible version for the last Expo SDK version (SDK 46 right now).

Any workaround?
If the fix is small, I could consider creating a patch for this.

**Comment by BrodaNoel** [comment]
If I manually apply [these changes](https://github.com/react-native-maps/react-native-maps/commit/e035c8dc16361412f9b42674e96bca152a158d99) and create a patch, it should work, right?

**Comment by BrodaNoel** [comment]
I guess only @Simon-TechForm may be able to answer this. Sorry for the mention!

**Comment by shashi-patel** [comment]
After upgrading to React Native version 0.77.0 the onMapReady event in react-native-maps is triggered only once when the MapView is initially rendered. However it does not trigger again after:

- Unmounting and remounting the screen
- Closing and reopening the app
- Navigating back to the page where MapView is used

This issue persists even after forcing a re-render by changing the key of the MapView.

**Expected Behavior:**
The onMapReady event should trigger each time the MapView is mounted including after navigating away and returning to the screen.
